google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -822,11 +832,49 @@ private void put(SpannerException e) {
     }
 
     private PooledSession take() throws SpannerException {
-      SessionOrError s = Uninterruptibles.takeUninterruptibly(waiter);
-      if (s.e != null) {
-        throw newSpannerException(s.e);
+      long currentTimeout = options.getInitialWaitForSessionTimeoutMillis();
+      while (true) {
+        Tracer tracer = Tracing.getTracer();
+        Scope waitScope = tracer.spanBuilder(WAIT_FOR_SESSION).startScopedSpan();
+        try {
+          SessionOrError s = pollUninterruptiblyWithTimeout(currentTimeout);
+          if (s == null) {
+            // End the span with status DEADLINE_EXCEEDED and retry.
+            tracer
+                .getCurrentSpan()
+                .end(EndSpanOptions.builder().setStatus(Status.DEADLINE_EXCEEDED).build());
```

Comment:
```suggestion
                .setStatus(Status.DEADLINE_EXCEEDED);
```

You don't need to end the span here. When scope is closed, span will be ended automatically. See https://github.com/census-instrumentation/opencensus-java/blob/916adcbaba580cb223cb37ca34cf1f902903ede9/api/src/main/java/io/opencensus/trace/CurrentSpanUtils.java#L97-L103.

Reply:
Changed.